Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs

```diff
@@ -0,0 +1,459 @@
+            
+            // Verify it's dirty
+            ref CharacterTransform dirtyTransform = ref globalWorld.Get<CharacterTransform>(avatarEntity);
+            Assert.IsTrue(dirtyTransform.IsDirty, "Transform should be dirty before system update");
+            
+            // Act - Run partition system
+            system.Update(0.1f);
+            
+            // Assert - Dirty flag should be cleared
+            ref CharacterTransform clearedTransform = ref globalWorld.Get<CharacterTransform>(avatarEntity);
+            Assert.IsFalse(clearedTransform.IsDirty, 
+                "PartitionSystem should clear IsDirty for avatar entities");
+        }
+
+        [Test]
+        public void OnlyProcessDirtyEntitiesWhenCameraNotDirty()
+        {
+            // Arrange
+            testGameObject = new GameObject("TestObject");
+            var dirtyTransform = new CharacterTransform(testGameObject.transform);
+            
+            GameObject cleanGameObject = new GameObject("CleanObject");
+            var cleanTransform = new CharacterTransform(cleanGameObject.transform);
+            
+            // Create two entities - one dirty, one clean
+            Entity dirtyEntity = globalWorld.Create(
+                dirtyTransform,
+                new PBAvatarShape(),
+                new PartitionComponent { Bucket = 0, IsBehind = false, IsDirty = false }
+            );
+            
+            Entity cleanEntity = globalWorld.Create(
+                cleanTransform,
+                new PBAvatarShape(), 
+                new PartitionComponent { Bucket = 0, IsBehind = false, IsDirty = false }
+            );
+            
+            // Make only one transform dirty
+            dirtyTransform.SetPositionWithDirtyCheck(Vector3.one * ABOVE_MINIMAL_DISTANCE_DIFFERENCE);
+            globalWorld.Set(dirtyEntity, dirtyTransform);
+            
+            // Verify states
+            ref CharacterTransform dirtyCheck = ref globalWorld.Get<CharacterTransform>(dirtyEntity);
+            ref CharacterTransform cleanCheck = ref globalWorld.Get<CharacterTransform>(cleanEntity);
+            Assert.IsTrue(dirtyCheck.IsDirty, "First entity should be dirty");
+            Assert.IsFalse(cleanCheck.IsDirty, "Second entity should be clean");
+            
+            // Act - Update with camera not dirty (should only process dirty entities)
+            cameraSamplingData.IsDirty.Returns(false);
+            system.Update(0.1f);
+            
+            // Assert
+            ref CharacterTransform dirtyAfter = ref globalWorld.Get<CharacterTransform>(dirtyEntity);
+            ref CharacterTransform cleanAfter = ref globalWorld.Get<CharacterTransform>(cleanEntity);
+            
+            Assert.IsFalse(dirtyAfter.IsDirty, "Dirty entity should be cleared");
+            Assert.IsFalse(cleanAfter.IsDirty, "Clean entity should remain clean");
+            
+            // Cleanup
+            Object.DestroyImmediate(cleanGameObject);
```

Comment:
You shouldn't clean-up in the test itself: if `Assert` fails, the rest of the code won't be executed: you've done it correctly with `testGameObject`: consider sticking to the same principle, you can have a list of objects if needed

Reply:
Thanks